Add example: write a 32-bit float value register (#19298)
home-assistant/home-assistant.github.io: source/_integrations/modbus.markdown

| value     | (write_register) A single value or an array of 16-bit values. Single value will call modbus function code 0x06. Array will call modbus function code 0x10. Values might need reverse ordering. E.g., to set 0x0004 you might need to set `[4,0]`, this depend on the byte order of your CPU |
| state     | (write_coil) A single boolean or an array of booleans. Single boolean will call modbus function code 0x05. Array will call modbus function code 0x0F |

### Example: writing a float32 type register

> To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Do:

Reviewer: ```suggestion
To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). 
```

No need to a action word here, the header and content below is enough

> To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Call the modbus.write_register:

Reviewer: ```suggestion
To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Call the `modbus.write_register` service:
```

Reviewer 2: I would prefer:
```suggestion
To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Do:
```
Because the content says directly it is a service call.

> To write a float32 datatype register with value decimal 10 first determine the 32 bit float value (`10` decimal == `0x41200000` in float hexadecimal notation). Then call the modbus.write_register as such:

Reviewer 2: ```suggestion
To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal). Call the modbus.write_register:
```

```yaml
service: modbus.write_register
data:
  address: <target register address>

>   unit: <target slave address>
>   hub: <hub name>

Reviewer: ```suggestion
```
When defining examples with placeholders only show required fields

Reviewer 2: But unit is required, and hub should be (it about to be corrected in an upcoming PR).
```
SERVICE_WRITE_REGISTER_SCHEMA = vol.Schema(
    {
        vol.Optional(ATTR_HUB, default=DEFAULT_HUB): cv.string,
        vol.Required(ATTR_UNIT): cv.positive_int,
        vol.Required(ATTR_ADDRESS): cv.positive_int,
        vol.Required(ATTR_VALUE): vol.Any(
            cv.positive_int, vol.All(cv.ensure_list, [cv.positive_int])
        ),
    }
)
```

Reviewer: If that is required, the documentation is wrong.

> unit: Slave address (1-255, defaults to 0)

Reviewer 2: you got me ! ! I will make a PR that updates the documentation . If hope you can accept that since it correspond with the code.

If you can accept it please approve.

Reviewer 2: please see https://github.com/home-assistant/home-assistant.io/pull/19313
